content/articles/build-autocorrect-feauture-using-nlp-with-python/index.md

```diff
@@ -0,0 +1,285 @@
+      `DELETE` - remove a letter.
+      `SWAP` - swap to adjacent letters.
+      `REPLACE` - changes one letter to another.
+  
+  >NOTE: We usually take n between 1 to 3 edits for autocorrect systems.
+
+ - *Filtering Suggested Candidates* - Only correctly spelt real words from our created candidate list are considered, so we can compare them to the words in the dictionary and then filter out the ones that don't exist in the dictionary.
+ - *Order Filtered Candidates based on word probabilities* - Probabilities of the words is calculated based on the following formula: P(w) = C(w)/V, where;
+      `P(w)`- the probability of a word,w.
+      `C(w)` - number of times(frequency) word appears in the vocabulary dictionary.
+      `V` - the total sum of words in the dictionary.
+ - *Choose the most Likely candidate* - When the probabilities are calculated, the actual list of words is used  to get the most likely word from our created candidates.
+
+### Build the Autocorrect Feature
+We will need a dictionary to develop the autocorrect system, much as our smartphone use prior history to match entered words whether they are correct or not. For this tutorial, we will use a [sample.txt](https://github.com/dentex22/Autocorrect_System/blob/main/sample.txt) file found in the project folder containing the 1000 most used vocabularies.
+
+#### Installing Libraries
+We start by installing all the libraries general to machine learning using `pip` command from the terminal.
+```python
+pip install pattern
+pip install pyspellchecker
+pip install autocorrect
+pip install textblob
+pip install textdistance
+```
+#### Reading text file(dictionary)
+We import all the necessary libraries and packages then read the text file containing our vocabulary dictionary with the code below.
+```python
+# Step 1: Data Preprocessing
+import re  # regular expression
+from collections import Counter
+import numpy as np
+import pandas as pd
+
+
+# Implement the function process_data which
+# 1) Reads in a corpus
+# #2) Changes everything to lowercase
+# 3) Returns a list of words.
+
+
+words = []
+with open('sample.txt','r',encoding="utf8") as f:
+    file_name_data = f.read()
+    file_name_data = file_name_data.lower()
+    words = re.findall('\w+', file_name_data)
+
+
+vocab = set(words)
+print(f"The first ten words in the text are: \n{words[0:10]}")
+print(f"There are {len(vocab)} unique words in the vocabulary.")
+```
+The following output is displayed in the terminal showing  number of unique words and the top 10 words in the dictionary. 
+
+```bash
+The first ten words in the text are: 
+['a', 'ability', 'able', 'about', 'above', 'accept', 'according', 'account', 'across', 'act']
+There are 1001 unique words in the vocabulary.
+```
+#### Frequency and Probability of Dictionary Words 
+We use the `counter function` to find the frequency of the words by applying the code below.
+```python
+# a get_count function that returns a dictionary of word vs frequency
```

Comment:
Change to " # a get_count function that returns a dictionary of word versus frequency"